modules/mapper-extras/src/main/java/org/opensearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -207,6 +207,9 @@ public SearchAsYouTypeFieldMapper build(Mapper.BuilderContext context) {
             // set up the prefix field
             FieldType prefixft = new FieldType(fieldType);
```

Comment:
I think this is what we want to change? `new FieldType(fieldType)` is copy ctor. I don't think we want to copy the defaults here.. maybe try: 
```suggestion
            FieldType prefixft = new FieldType();
            prefixft.setIndexOptions(fieldType.indexOptions());
```

Which will get the index options set by the user in `index_options`.

Reply:
I did your suggested changes and added tests.